client/src/pages/ScanStore.tsx

```diff
@@ -47,39 +91,259 @@ function ScanStore() {
     setShoppingList(sList);
   };
 
+  const setDebugItem = (barcode: string) => {
+    setCurBarcode(barcode);
+    setShoppingList(cartItems);
+  };
+
+  const addItem = async () => {
+    if (curBarcode == "") {
+      // If empty, use media API
+      const imgReq = {
+        allowedMimeTypes: ["image/jpeg"],
+        allowedSources: ["camera"], // Restrict to camera scanning only
+      };
+      const imgRes = await window.microapps
+        .requestMedia(imgReq)
+        .then((res: any) => res);
+      setUploadImg(imgRes);
+    } else {
+      let barcodes: string[] = [curBarcode];
+      fetchItem(barcodes);
+    }
+  };
+
+  const fetchItem = async (barcodes: string[]) => {
+    // Check first if we have previously retrieved this item
+    let unknown_barcodes: string[] = [];
+    let matchedItems: Item[] = [];
+    for (let i = 0; i < barcodes.length; ++i) {
+      if (idxMap[barcodes[i]] != undefined) {
```

Comment:
Try to put comments for each if/else case to help other developers understand the different approach for different cases.

Reply:
#30